server/backend/api/alembic/versions/V18_add_role_app_constraint_and_app_env_attr.py

```diff
@@ -0,0 +1,89 @@
+"""Add Role-App Constraint and App-Environment Attribute
+
+Revision ID: V18
+Revises: V17
+Create Date: 2023-01-17 16:00:58.613953
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import sqlmodel
+
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = 'V18'
+down_revision = 'V17'
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    # Environment Type Code.
+    fam_app_environment_type_table=op.create_table('fam_app_environment_type',
+    sa.Column('app_environment_type_code', sa.String(length=4), nullable=False, comment='application environment type code'),
+    sa.Column('description', sa.String(length=100), nullable=True, comment='Description of what the app_environment_type_code represents.'),
+    sa.Column('effective_date', postgresql.TIMESTAMP(precision=6), server_default=sa.text('now()'), nullable=False, comment='The date and time the code was effective.'),
+    sa.Column('expiry_date', postgresql.TIMESTAMP(precision=6), nullable=True, comment='The date and time the code expired.'),
```

Comment:
What about create/update user fields?
Why have an expiry_date column if we're not planning to use it (I haven't looked at the rest of the code changes yet though).

Reply:
As discussed, will ignore this one as it needs more backlog refinement to address the design on several tables for this issue. Here simply just following through.